#!/usr/bin/env python
PACKAGE = "pose_cnn"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("icp_max_correspondence_distance", double_t, 0, "Max Correspondence Distance for ICP",    0.02, 0,   1)
gen.add("icp_transformation_epsilon",    double_t,    0, "Transformation Epsilon for ICP", 1e-9, 0,   1)
gen.add("icp_euclidean_fitness_epsilon",   double_t,   0, "Euclidean Fitness Epsilon for ICP",  5e-5, 0, 1)
gen.add("icp_iterations",   int_t,   0, "Max ICP iterations",  100, 0, 10000)

gen.add("voxel_grid_leaf_size", double_t, 0, "Leaf size for Voxel grid sampling.", 0.05, 0, 1)

gen.add("ransac_distance",    double_t,    0, "Inlier distance for RANSAC", 0.01, 0,   0.2)

exit(gen.generate(PACKAGE, "pose_cnn", "PoseCNNRefined"))